File: Solutions/1664. Ways to Make a Fair Array/1664. Ways to Make a Fair Array.cpp
class Solution {
public:
    int waysToMakeFair(vector<int>& nums) {
        int res=0,l[2]={},r[2]={};
        for(int i=0;i<nums.size();++i)
            r[i%2]+=nums[i];
        for(int i=0;i<nums.size();++i){
            r[i%2]-=nums[i];
            res+=l[0]+r[1]==l[1]+r[0];
            l[i%2]+=nums[i];
            }
        return res;
    }
}; 

File: Solutions/1664. Ways to Make a Fair Array/1664. Ways to Make a Fair Array.py
class Solution:
    def waysToMakeFair(self, nums: List[int]) -> int:
        l,r,res=[0,0],[0,0],0
        for i in range(len(nums)):
            r[i%2]+=nums[i]
        for i in range(len(nums)):
            r[i%2]-=nums[i]
            x,y=r[0]+l[1],r[1]+l[0]
            if x==y:
                res+=1
            l[i%2]+=nums[i]
        return res